src/proto/client/mod.rs

```diff
@@ -359,10 +373,10 @@ where
         Ok((jobs_count - errors.len(), Some(errors)))
     }
 
-    /// Retrieve information about the running server.
+    /// Retrieve [information](crate::ServerState) about the running server.
     ///
     /// The returned value is the result of running the `INFO` command on the server.
-    pub async fn info(&mut self) -> Result<serde_json::Value, Error> {
+    pub async fn info(&mut self) -> Result<single::ServerState, Error> {
```

Comment:
This is breaking and if we decide it's worth it  - these changes should join the upcoming breaking 'async' release.
We expose an _additional_ method like `get_server_state` which would be returning `ServerState`, but this would be confusing for the lib user.